src/CMakeLists.txt

```diff
@@ -0,0 +1,246 @@
+)
+
+# Mayaqua kernel
+file(GLOB SOURCES_MAYAQUA "Mayaqua/*.c")
+file(GLOB HEADERS_MAYAQUA "Mayaqua/*.h")
+
+add_library(mayaqua STATIC ${SOURCES_MAYAQUA} ${HEADERS_MAYAQUA})
+
+target_include_directories(mayaqua PUBLIC Mayaqua)
+
+set_target_properties(mayaqua
+  PROPERTIES
+  ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/Mayaqua"
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/Mayaqua"
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/Mayaqua"
+)
+
+# hamcorebuilder utility
+add_executable(hamcorebuilder hamcorebuilder/hamcorebuilder.c)
+
+set_target_properties(hamcorebuilder
+  PROPERTIES
+  ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp"
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp"
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp"
+)
+
+# vpnserver
+add_executable(vpnserver vpnserver/vpnserver.c)
+
+set_target_properties(vpnserver
+  PROPERTIES
+  ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/vpnserver"
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpnserver"
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpnserver"
+)
+
+get_target_property(VPNSERVER_RUNTIME_OUTPUT_DIRECTORY vpnserver RUNTIME_OUTPUT_DIRECTORY)
+
+# vpnclient
+add_executable(vpnclient vpnclient/vpncsvc.c)
+
+set_target_properties(vpnclient
+  PROPERTIES
+  ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/vpnclient"
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpnclient"
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpnclient"
+)
+
+get_target_property(VPNCLIENT_RUNTIME_OUTPUT_DIRECTORY vpnclient RUNTIME_OUTPUT_DIRECTORY)
+
+# vpnbridge
+add_executable(vpnbridge vpnbridge/vpnbridge.c)
+
+set_target_properties(vpnbridge
+  PROPERTIES
+  ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/vpnbridge"
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpnbridge"
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpnbridge"
+)
+
+get_target_property(VPNBRIDGE_RUNTIME_OUTPUT_DIRECTORY vpnbridge RUNTIME_OUTPUT_DIRECTORY)
+
+# vpncmd
+add_executable(vpncmd vpncmd/vpncmd.c)
+
+set_target_properties(vpncmd
+  PROPERTIES
+  ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/tmp/vpncmd"
+  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpncmd"
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin/vpncmd"
+)
+
+get_target_property(VPNCMD_RUNTIME_OUTPUT_DIRECTORY vpncmd RUNTIME_OUTPUT_DIRECTORY)
+
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  add_definitions(-D_DEBUG -DDEBUG)
+endif()
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+  add_definitions(-DNDEBUG -DVPN_SPEED)
+endif()
+
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    add_definitions(-DCPU_64)
+endif()
+
+add_definitions(-D_REENTRANT -DREENTRANT -D_THREAD_SAFE -D_THREADSAFE -DTHREAD_SAFE -DTHREADSAFE -D_FILE_OFFSET_BITS=64)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsigned-char")
+
+# Add /src to the include paths
+include_directories(.)
+
+if(WIN32)
+  message(FATAL_ERROR "Windows compilation via CMake is currently not supported.")
```

Comment:
Should't the program exit here?

Reply:
From [CMake's documentation](https://cmake.org/cmake/help/v3.0/command/message.html):
`FATAL_ERROR    = CMake Error, stop processing and generation`